utils/src/main/java/com/cloud/utils/ssh/SshHelper.java

```diff
@@ -207,22 +206,18 @@ public static void scpTo(String host, int port, String user, File pemKeyFile, St
                     if (canEndTheSshConnection(waitResultTimeoutInMs, sess, conditions)) {
                         break;
                     }
-
                 }
 
-                while (stdout.available() > 0) {
-                    currentReadBytes = stdout.read(buffer);
-                    sbResult.append(new String(buffer, 0, currentReadBytes));
+               while((currentReadBytes = stdout.read(buffer)) > 0) {
```

Comment:
@Pearl1594 instead of `> 0` we probably want to check for readbytes `!= -1`. For example: (comments only for your understanding)
```
currentReadBytes = stdout.read(buffer);
while (currentReadBytes >= 0) { // or check for != -1
  sbResult.append(new String(buffer, 0, currentReadBytes));
  currentReadBytes = stdout.read(buffer); // read() will return -1 on eof/complete
}
```
The other option is to use a more modern java-way to use `BufferedReader` for example like:
```
 BufferedReader bufferedStdOut = new BufferedReader(stdout);
.... use
```

Reply:
@rhtyd I did consider using that, however decided against it so at to avoid wrapping the Input Stream. Also used the readAllBytes() but that would increase the memory footprint as indicated earlier. So, went ahead with this approach.

Reply:
@Pearl1594 long shot but you are using a test for '> 0'. What if the stream is blocking due to timing issues but not ended yet? Wouldn't we still end reading before end of stream?

Reply:
never mind, guess you already changed that.